### PR TITLE
added test for long file name

### PR DESCRIPTION
**Pull request by sushmita56**
### Related Issue
#9396 
### Description
The file name with length(>=233 character) is not synced and is blacklisted.

Screenshot from 2022-01-28 15-26-33

This PR has been closed due to some reason, so instead of this another PR is created here->#9458.

### CODE REVIEW

File: test/gui/tst_syncing/test.feature [hunk 1]
```diff
@@ -292,3 +292,27 @@ Feature: Syncing files
...
+
+
+    Scenario: File with long name can be synced
+        Given user "Alice" has set up a client with default settings
+        # the length of file name is 224
+        When user "Alice" creates a file "sdkdfjsdfuidjfkdsjfksdjfksdjfksdjfksdjfksdjfkdsjfksdjfkdsjfkdsfjlsdkfjsdkjflksdjfklsdjfksdjfkdsjfkldsjfkldsjfkdsjfksdjfksdjfklsdjfklsdjflksdjflksdjfklsdjfklsdjfksdjfksdjfksdjfksdjfksdfjskdfjksdjfksdjfksdjfksdjfksdwwwwww.txt" with the following content inside the sync folder
+            """
+            test content
+            """
+        And the user waits for file "sdkdfjsdfuidjfkdsjfksdjfksdjfksdjfksdjfksdjfkdsjfksdjfkdsjfkdsfjlsdkfjsdkjflksdjfklsdjfksdjfkdsjfkldsjfkldsjfkdsjfksdjfksdjfklsdjfklsdjflksdjflksdjfklsdjfklsdjfksdjfksdjfksdjfksdjfksdfjskdfjksdjfksdjfksdjfksdjfksdwwwwww.txt" to be synced
```
Comment (Talank) [inline test/gui/tst_syncing/test.feature:304]: Please use example table instead of repeating long name

File: test/gui/shared/steps/steps.py [hunk 2]
```diff
@@ -331,8 +332,9 @@ def collaboratorShouldBeListed(
 
 @When('the user waits for the files to sync')
 def step(context):
+    snooze(20)
     waitForFolderToBeSynced(context, '/')
-
+    snooze(30)
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:337]: You probably won't need these snooze 🤔

File: test/gui/shared/steps/steps.py [hunk 3]
```diff
@@ -378,8 +380,7 @@ def createFile(context, filename, username=None):
     f = open(join(syncPath, filename), "w")
     f.write(fileContent)
     f.close()
-
-
+
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:383]: please revert this change back to 2 empty lines between the two step definitions/functions

File: test/gui/shared/steps/steps.py [hunk 4]
```diff
@@ -399,14 +400,63 @@ def createFolder(context, foldername, username=None):
     path = join(syncPath, foldername)
     os.makedirs(path)
 
-
+ 
+@When('user "|any|" creates "|any|" files inside the folder "|any|"')
+def step(context, username, filenumber, foldername):
+    syncPath = context.userData['currentUserSyncPath']
+    path = join(syncPath, foldername)
+    snooze(5)
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:408]: Add comment for the reason to put snooze here. The reason is probably the same as that in create file. Nevertheless, write the comment as well.

File: test/gui/shared/steps/steps.py [hunk 5]
```diff
@@ -399,14 +400,63 @@ def createFolder(context, foldername, username=None):
...
+    for i in range(0, builtins.int(filenumber)):
+        file_name = f"file{i}.txt"
+        file = open(f'{path}/file{i}', "wb")
+        file.seek(1048576)
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:412]: We might want flexibility on the size of file. So instead of hardcoding the number, can you put it in some variable. It would be nice if you can get this data from the feature file.

File: test/gui/shared/steps/steps.py [hunk 6]
```diff
@@ -399,14 +400,63 @@ def createFolder(context, foldername, username=None):
...
+        file.write(b"\0")
+        file.close()
+
+
+@When('user "|any|" creates a "|any|" subfolders inside the folder "|any|"')
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:417]:
```suggestion
@When('user "|any|" creates "|any|" subfolders inside the folder "|any|"')
```

File: test/gui/shared/steps/steps.py [hunk 7]
```diff
@@ -399,14 +400,63 @@ def createFolder(context, foldername, username=None):
...
+@Then('pause')
+def step(context):
+    snooze(60)
+    
+    
+@When('pause')
+def step(context):
+    snooze(60)
+    
+
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:459]: Remove these steps

**Review by Talank — CHANGES_REQUESTED** [review]

File: test/gui/tst_syncing/test.feature [hunk 8]
```diff
@@ -276,19 +277,116 @@ Feature: Syncing files
...
+        When the user waits for the files to sync
+        Then as "Alice" folder "folder1" should exist on the server
+        And as user "Alice" folder "folder1" should contain "500" items on the server
+        When the user waits for the files to sync
+        Then as "Alice" folder "folder2" should exist on the server
+        And as user "Alice" folder "folder2" should contain "500" items on the server
+
+
+    
+    Scenario: Folders with 15 subfolders can sync successfully
```
Comment (Talank) [inline test/gui/tst_syncing/test.feature:378]: keep 2 spaces between scenarios

File: test/gui/shared/steps/steps.py [hunk 9]
```diff
@@ -399,14 +399,57 @@ def createFolder(context, foldername, username=None):
...
-
+ 
+@When('user "|any|" creates "|any|" files each of size "|any|" bytes inside the folder "|any|"')
+def step(context, username, filenumber, filesize, foldername):
+    syncPath = context.userData['currentUserSyncPath']
+    path = join(syncPath, foldername)
+    filesize = builtins.int(filesize)
+    # Syncing all the created files at once seems to have some problem on syncing
+    # So once a file is synced the process have to wait unless the next file is created
+    # So we added Snooze of 5 sec
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:410]:
```suggestion
    # A file is scheduled to be synced but is marked as ignored for 5 seconds. And if we try to sync it, it will fail. So we need to wait for 5 seconds.
    # https://github.com/owncloud/client/issues/9325
```

File: test/gui/shared/steps/steps.py [hunk 10]
```diff
@@ -399,14 +399,57 @@ def createFolder(context, foldername, username=None):
...
+    for i in range(builtins.int(folder_count)):
+        sub_folder ='sub{i}'.format(i=i)
+        path = join(syncPath, folder, sub_folder)
+        os.mkdir(path)
+    
+
+@When('user "|any|" creates a file "|any|" with size "|any|" inside the sync folder')
+def step(context, username, filename, filesize):
+    uploadFile(context, username, filename, filesize)
+
```
Comment (Talank) [inline test/gui/shared/steps/steps.py:432]:
```suggestion


```
To be consistent

File: test/gui/tst_syncing/test.feature [hunk 11]
```diff
@@ -279,16 +280,112 @@ Feature: Syncing files
...
+        And user "Alice" creates "100" subfolders inside the folder "folder2"
+        And the user waits for folder "folder1" to be synced
+        Then as "Alice" folder "folder1" should exist on the server
+        And as user "Alice" folder "folder1" should contain "100" items on the server
+        When the user waits for folder "folder2" to be synced
+        Then as "Alice" folder "folder2" should exist on the server
+        And as user "Alice" folder "folder2" should contain "100" items on the server
+
+
+
```
Comment (Talank) [inline test/gui/tst_syncing/test.feature:391]:
```suggestion
```
Remove unnecessary spaces at the end

File: test/gui/shared/steps/steps.py (hunk 4 above)
Comment (sushmita56) [reply test/gui/shared/steps/steps.py:407]: Syncing all the created files at once seems to have some problem on syncing, as a result not all created files can be synced. So once a file is synced the process will have to wait unless the next file is created. So here we added a snooze of 5 seconds.

Also, the step ``` When the user waits for the files to sync ``` is not working properly as expected. And probably this PR -> #9442 may solve this issue. So we can’t remove ```snooze(5)``` unless the issue is fixed.

File: test/gui/tst_syncing/test.feature [hunk 12]
```diff
@@ -329,4 +330,97 @@ Feature: Syncing files
...
+        # the length of file name is 224
+        When user "Alice" creates a file "<filename>" with the following content inside the sync folder
+            """
+            test content
+            """
+        And the user waits for file "<filename>" to be synced
+        Then as "Alice" file "<filename>" should exist on the server
+        Examples:
+            | filename                                                                                                                                                                                                                        |
+            | sdkdfjsdfuidjfkdsjfksdjfksdjfksdjfksdjfksdjfkdsjfksdjfkdsjfkdsfjlsdkfjsdkjflksdjfklsdjfksdjfkdsjfkldsjfkldsjfkdsjfksdjfksdjfklsdjfklsdjflksdjflksdjfklsdjfklsdjfksdjfksdjfksdjfksdjfksdfjskdfjksdjfksdjfksdjfksdjfksdwwwwww.txt |
```
Comment (Talank) [inline test/gui/tst_syncing/test.feature:347]: can you create a name a bit more readable?

File: test/gui/tst_syncing/test.feature [hunk 13]
```diff
@@ -329,4 +330,97 @@ Feature: Syncing files
...
+            test contents
+            """
+        When user "Alice" creates a file "file1.odt" with the following content inside the sync folder
+            """
+            test contents
+            """
+        And the user waits for file "file1.txt" to be synced
+        And the user waits for file "file1.odt" to be synced
+        Then as "Alice" file "file1.txt" should exist on the server
+        Then as "Alice" file "file1.odt" should exist on the server
```
Comment (Talank) [inline test/gui/tst_syncing/test.feature:379]:
```suggestion
        And as "Alice" file "file1.odt" should exist on the server
```

**Comment by jnweiger** [comment]
@sushmita56 was this intentionally closed?

**Comment by phil-davis** [comment]
See PR #9458
